sonic-post-syseeprom/scripts/post-syseeprom

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python2
+
+'''
+    post-syseeprom
+    Syseeprom information gathering tool for SONiC
+    This tool will be started during the start phase of pmon container, gathering syseeprom info and write to state DB.
+    It's an one-shot task since syseeprom info are static.
+    With this tool, show syseeprom CLI will be able to get data from state DB instead of access hw or cache.
+'''
+
+try:
+    from sonic_daemon_base.daemon_base import DaemonBase
+    from sonic_daemon_base.daemon_base import Logger
+    from sonic_daemon_base import daemon_base
+    import argparse
+    from swsscommon import swsscommon
+except ImportError, e:
+    raise ImportError (str(e) + " - required module not found")
+
+PLATFORM_SPECIFIC_MODULE_NAME = 'eeprom'
+PLATFORM_SPECIFIC_CLASS_NAME = 'board'
+
+ERR_ARISTA_PLATFORM = 1
+ERR_FAILED_EEPROM = 2
+ERR_FAILED_UPDATE_DB = 3
+
+EEPROM_TABLE_NAME = 'EEPROM_INFO'
+SYSLOG_IDENTIFIER = 'post-syseeprom'
+
+# Global logger class instance
+logger = Logger(SYSLOG_IDENTIFIER)
+
+def main():
+    parser = argparse.ArgumentParser(description='post-syseeprom tool')
+    parser.add_argument("-w", help="write syseeprom info into state DB", action='store_true')
+    parser.add_argument("-c", help="remove syseeprom info from state DB", action='store_true')
+    args = parser.parse_args()
+
+    Daemon_Base = DaemonBase()
+    (platform_path, hwsku_path) = Daemon_Base.get_path_to_platform_and_hwsku()
+
+    # Arista platform have their own implementation for eeprom tlv parsing, it's not inherited from eeprom base class.
+    # So for now this tool can not support Arista platform, but this will not cause any issue on them. 
+    # decode-syseeprom scripts will cover Arista case, it will still read from HW/cache instead from DB on Arista platform.  
+    if 'arista' in platform_path:
+        logger.log_warning("Arista platform does not support this yet")
+        return ERR_ARISTA_PLATFORM
+
+    if args.w:
+        eeprom_util = Daemon_Base.load_platform_util(PLATFORM_SPECIFIC_MODULE_NAME, PLATFORM_SPECIFIC_CLASS_NAME)
+        eeprom = eeprom_util.read_eeprom()
+        if eeprom is None :
+            logger.log_error("Failed to read eeprom")
+            return ERR_FAILED_EEPROM
+
+        err = eeprom_util.update_eeprom_db(eeprom)
```

Comment:
>update_eeprom_db [](start = 26, length = 16)

Are you going to write MAC address into StateDB? If yes, there is already one in ConfigDB, how do you handle data consistency?

Reply:
The purpose here is to put all the syseeprom info into state DB and afterward, all the request from CLI or somewhere else to check the syseeprom info will access the state DB instead of reading the from the hardware.  The one in the config DB may or may not consistent with the one burnt to syseeprom and depends on the implementaion of specific platform?  anyway these info is only for show not for other purpose.

Reply:
Please check old PR: [https://github.com/Azure/sonic-buildimage/pull/1397](https://github.com/Azure/sonic-buildimage/pull/1397)

Digging into the history, we introduced 'DEVICE_METADATA|localhost :: mac' only for Mellanox platform, because other platform could easily get the mac from eth0.

Here you introduce a duplication of mac in Redis database. Could you please consolidate into one?

---
In reply to: [284053727](https://github.com/Azure/sonic-platform-daemons/pull/25#discussion_r284053727) [](ancestors = 284053727)

Reply:
I know there is lots of twist on how to get the switch mac address. 

- The mac address in config db is essential to the start of syncd(or swss?), need to be fetched from syseeprom at an early stage of switch starting, in this PR. the syseeprom info(including the mac address) is read in a later phase(at the starting of the pmon), so we can not feed the mac address in config DB with this scripts.

- on the other hand, if we want to skip the mac address in this script and try to reuse the mac address in config db, then we need to refactor the "decode-syseeprom" code to get the base mac address from another source, that will cause the code more complicated, I am thinking the benefit what we can get from reducing the mac address here is not so significant.

- you mentioned that on other platforms just directly use the "eth0" as the mac address,  in this scripts the mac adress read from syseeprom is the base mac, usually they are different from the "eth0" mac address.

Reply:
I think you still need `return main()`

---
In reply to: [283998899](https://github.com/Azure/sonic-platform-daemons/pull/25#discussion_r283998899) [](ancestors = 283998899)

Reply:
I don't get the point here. However, I unblocked this PR and try to discuss offline and get better understanding of the motivation.

---
In reply to: [285456136](https://github.com/Azure/sonic-platform-daemons/pull/25#discussion_r285456136) [](ancestors = 285456136)

Reply:
> I think you still need `return main()`
> 
> In reply to: [283998899](https://github.com/Azure/sonic-platform-daemons/pull/25#discussion_r283998899) [](ancestors = 283998899)

amended.